docs/Using Randomness/chainlink-vrf.md

```diff
@@ -4,30 +4,108 @@ section: ethereum
+
+The Subscription Manager system has the following limitations:
+
+- Each subscription supports up to 100 consumer contracts. If you need more than 100 consumers, create multiple subscriptions.
+- Each subscription must maintain a sufficient balance to fund requests from consumer contracts. If your balance cannot cover the cost to complete a request, the request remains pending. Top up the subscription balance with LINK. The request will go through automatically as long as you made the request in the last 24 hours.
+
+## Subscription billing
+
+For Chainlink VRF v2 to fulfill your requests, you must maintain a sufficient amount of LINK in your subscription balance. Gas cost calculation includes the following variables:
+
+- **Gas price:** The current gas price, which fluctuates depending on network conditions
+
+- **Callback gas:** The amount of gas used for the callback request that returns your requested random values
+
+- **Verification gas:** The amount of gas used to verify randomness on-chain
+
+These variables depend on current network conditions, your specified limit on callback gas, and the number of random values in your request. The cost of each request is final only after the transaction is complete, but you define the limits you are willing to spend for the request with the following variables:
+
+- **Gas lane:** The maximum gas price you are willing to pay for a request in wei. Define this limit by specifying the appropriate `keyHash` in your request. The limits of each gas lane are important for handling gas price spikes when Chainlink VRF bumps the gas price to fulfill your request quickly.
+
+- **Callback gas limit:** Specifies the maximum amount of gas you are willing to spend on the callback request. Define this limit by specifying the `callbackGasLimit` value in your request.
+
+Requests to Chainlink VRF v2 follow the [Request & Receive Data](/docs/request-and-receive-data/) cycle. The VRF coordinator processes the request and determines the final charge to your subscription using the following steps:
+
+1. You submit your request with a specified gas lane `keyHash` and the `callbackGasLimit`. If your request is urgent, specify a gas lane with a higher gas price limit. The `callbackGasLimit` depends on the size of your request. Generally, a limit of 100,000 gas is appropriate. Verification gas has a hard upper limit of 200,000 gas.
+
+1. The coordinator starts the transaction and bumps the gas price until the transaction is completed. The coordinator will not exceed the gas price limit of your selected gas lane. If your request cannot be completed at your specified gas limit, resubmit the request with a different gas lane or wait until gas prices fall below your current gas lane limits.
```

Comment:
Is it intentional that all numbers in this numbered list are `1.`?

Reply:
Yes. Markdown auto-orders them. It's easier to insert and remove points if we don't manually manage the order.